### ISSUE TITLE
com.hazelcast.client.stress.MapUpdateStressTest.testFixedCluster

### ISSUE THREAD

**Issue by sumnerib** [hazelcast/hazelcast#22669]
_master_ (commit 8f8caf5580b7f05172c7323a14a22d48dab2f340)

Failed on OracleJDK8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-OracleJDK8-nightly/921/testReport/junit/com.hazelcast.client.stress/MapUpdateStressTest/testFixedCluster/
<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: com.hazelcast.client.stress.MapUpdateStressTest$StressThread16 encountered an error expected null, but was:<java.lang.NullPointerException>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotNull(Assert.java:756)
	at org.junit.Assert.assertNull(Assert.java:738)
	at com.hazelcast.client.stress.StressTestSupport$TestThread.assertNoError(StressTestSupport.java:187)
	at com.hazelcast.client.stress.StressTestSupport.assertNoErrors(StressTestSupport.java:138)
	at com.hazelcast.client.stress.StressTestSupport.joinAll(StressTestSupport.java:159)
	at com.hazelcast.client.stress.MapUpdateStressTest.test(MapUpdateStressTest.java:91)
	at com.hazelcast.client.stress.MapUpdateStressTest.testFixedCluster(MapUpdateStressTest.java:84)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:750)

```
</details>

Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-OracleJDK8-nightly/921/

**Comment by krzysztofslusarski** [comment]
```
java.lang.NullPointerException
        at com.hazelcast.client.stress.MapUpdateStressTest$StressThread.doRun(MapUpdateStressTest.java:151)
        at com.hazelcast.client.stress.StressTestSupport$TestThread.run(StressTestSupport.java:176)
```

**Comment by krzysztofslusarski** [comment]
The only way to create that exception is to make ```map = client.getMap("map");``` return ```null```. That nightly run has plenty of failures. I checked the code of that test, the map proxy is safely published to the stress thread, since it is created before starting thread and creation and starting is invoked from the same thread.

**Comment by krzysztofslusarski** [comment]
Full conversation here: https://hazelcast.slack.com/archives/C01JU7ZJYGP/p1668592206511029

TLDR: Stress test uses real network, we can see in the logs the member outside of the test that messing around with the cluster.
